File: BattleGame/backend/GetAssetsByPlayer.cs

using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Http;
using Microsoft.Data.SqlClient;
using System.Net;
using System.Text.Json;

public class GetAssetsByPlayer
{
    private readonly string conn = Environment.GetEnvironmentVariable("SqlConnection");

    [Function("getassetsbyplayer")]
    public async Task<HttpResponseData> Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequestData req)
    {
        var list = new List<object>();

        using SqlConnection connection = new SqlConnection(conn);
        connection.Open();

        string q = @"SELECT p.PlayerName,p.Level,p.Age,a.AssetName
                     FROM PlayerAsset pa
                     JOIN Player p ON pa.PlayerId=p.Id
                     JOIN Asset a ON pa.AssetId=a.Id";

        var cmd = new SqlCommand(q,connection);
        var reader = cmd.ExecuteReader();

        while(reader.Read())
        {
            list.Add(new {
                PlayerName = reader["PlayerName"],
                Level = reader["Level"],
                Age = reader["Age"],
                AssetName = reader["AssetName"]
            });
        }

        var res = req.CreateResponse(HttpStatusCode.OK);
        await res.WriteStringAsync(JsonSerializer.Serialize(list));
        return res;
    }
}

File: BattleGameAPI/src/functions/getassetsbyplayer.js
const { app } = require('@azure/functions');
const pool = require('../../db');

app.http('getassetsbyplayer', {
    methods: ['GET'],
    authLevel: 'anonymous',

    handler: async (request, context) => {

        try {

            const [rows] = await pool.query(`
                SELECT 
                p.PlayerName,
                p.Level,
                p.Age,
                a.AssetName
                FROM Player p
                JOIN PlayerAsset pa ON p.PlayerId = pa.PlayerId
                JOIN Asset a ON pa.AssetId = a.AssetId
            `);

            return {
                status: 200,
                jsonBody: rows
            };

        } catch (error) {

            return {
                status: 500,
                jsonBody: { error: error.message }
            };

        }
    }
});